### DISCUSSION TITLE
EBSD Sequence

### DISCUSSION THREAD

**Discussion by oMuransky** [mtex-toolbox/mtex#1933] [Ask Anything] [answered]
Hi Ralf, 

I don't seem to be able to find a similar discussion, but if it exists, it would be appreciated if you could direct me to it. 

In short, I seek a methodology for analysing a series of EBSD data sets, either simulated or measured. Specifically, I am interested in calculating the grains only from the initial map, which could be, for example, collected at zero stress/strain. Subsequent EBSD maps should inherit the grain assignments from this initial EBSD map. Is there an existing approach to accomplish this in MTEX?

Many thanks,
Ondrej

**Comment by ralfHielscher** [comment]
Hi Ondrej,

if the deformed EBSD maps have the same size as the initial EBSD map, you can simply do
```matlab
ebsdDef.grainId = ebsd.grainId;
```
Most likely you need also a transformation that brings the two EBSD maps into coincidence. 

Ralf.

**Reply by oMuransky** [reply]
Thank you for your prompt reply, Ralf. Since I'm working with simulated data, this would quickly sort out a few things. However, is there a way to force the calcGrains function to obey ebsd.grainId? I would still want access to grains properties like meanOrientation and mis2mean, which will require calculating the grains (if I understand things correctly). Ultimately, the idea is to plot e.g. meanOrientation of grains, as a function of the applied load and compare simulations with an experiment.

**Reply by ralfHielscher** [reply, marked as answer]
Yes, this should be possible using the option 
```matlab
grainsDef = calcGrains(ebsdDef,'custom',ebsd.grainId)
```
Ralf.

**Reply by oMuransky** [reply]
Wow. Fantastic! You thought of everything. This sorted all my problems. 

Many thanks,
Ondrej
